src/security/tpm/tspi/log-tpm2.c

```diff
@@ -22,23 +22,6 @@
 #include <cbmem.h>
 #include <vb2_sha.h>
 
-static uint16_t tpmalg_from_vb2_hash(enum vb2_hash_algorithm hash_type)
-{
-	switch (hash_type) {
-	case VB2_HASH_SHA1:
-		return TPM2_ALG_SHA1;
-	case VB2_HASH_SHA256:
-		return TPM2_ALG_SHA256;
-	case VB2_HASH_SHA384:
-		return TPM2_ALG_SHA384;
-	case VB2_HASH_SHA512:
-		return TPM2_ALG_SHA512;
-
-	default:
-		return 0xFF;
-	}
-}
-
 void *tpm2_log_cbmem_init(void)
```

Comment:
One more thing to that function. The log size should be 64K at minimum per the TCG spec:

```
4. If the ACPI TPM2 table contains the address and size of the Platform Firmware TCG log:
a. The Log Area Minimum Length for the TCG event log MUST be at least 64KB.
```

Here we just allocate 50 entries + size of the log table. Same happens of TPM1.2. Because of that the platform ends up having just a few hundreds of bytes of the log in ACPI TPM table. Not sure if measured boot v2 PRds change that on EDK2 side somehow...

Reply:
It does (discarding anything that coreboot wrote there), although it respects the length reported in ACPI.  Added the change anyway: https://github.com/Dasharo/coreboot/pull/699/commits/84fb089a040e938ced09b2d74d1b5dd9d17e3348.

Reply:
Thank you